api/src/routes/auth-dev.test.ts

```diff
@@ -0,0 +1,199 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import {
+  defaultUserEmail,
+  devLogin,
+  setupServer,
+  superRequest
+} from '../../jest.utils';
+import { HOME_LOCATION } from '../utils/env';
+
+describe('dev login', () => {
+  setupServer();
+
+  beforeEach(async () => {
+    await fastifyTestInstance.prisma.user.deleteMany({
+      where: { email: defaultUserEmail }
+    });
+  });
+
+  afterAll(async () => {
+    await fastifyTestInstance.prisma.user.deleteMany({
+      where: { email: defaultUserEmail }
+    });
+  });
+
+  describe('GET /signin', () => {
+    it('should create an account if one does not exist', async () => {
+      const res = await superRequest('/signin', { method: 'GET' });
+
+      const count = await fastifyTestInstance.prisma.user.count({
+        where: { email: defaultUserEmail }
+      });
+
+      expect(count).toBe(1);
+      expect(res.body).toStrictEqual({});
+      expect(res.status).toBe(302);
+    });
+
+    it('should populate the user with the correct data', async () => {
+      const uuidRe = /^[a-f0-9]{8}-([a-f0-9]{4}-){3}[a-f0-9]{12}$/;
+      const fccUuidRe = /^fcc-[a-f0-9]{8}-([a-f0-9]{4}-){3}[a-f0-9]{12}$/;
+
+      await superRequest('/signin', { method: 'GET' });
+      const user = await fastifyTestInstance.prisma.user.findFirstOrThrow({
+        where: { email: defaultUserEmail }
+      });
+
+      expect(user).toMatchObject({
```

Comment:
I realise this has just moved file. `toMatchObject` allows subsets to match. So, if we want to ensure all properties exist, we might want to use `toStrictEqual`.